src/python/WMCore/Services/ReqMgrAux/ReqMgrAux.py

```diff
@@ -57,14 +57,76 @@ def _getDataFromMemoryCache(self, callname):
         return cache.getData()
 
     def getCMSSWVersion(self):
+        """
+        get dictionary format of architecture and cmssw versions
+        i.e.
+        {"slc5_amd64_gcc462": [
+          "CMSSW_5_3_4_patch1",
+          "CMSSW_5_2_9",
+          "CMSSW_5_3_4_patch2",
+          "CMSSW_5_3_3_cand1_patch1"],
+          ...
+          }
+        """
         return self._getDataFromMemoryCache('cmsswversions')
 
+    def updateRecords(self, callname, kwparams):
+
+        return self["requests"].put(callname, kwparams)[0]['result']
 
     def populateCMSSWVersion(self, tc_url, **kwargs):
+        """
+        query TagCollector and populate cmsswversions
+        """
         from WMCore.Services.TagCollector.TagCollector import TagCollector
         cmsswVersions = TagCollector(tc_url, **kwargs).releases_by_architecture()
 
         return self["requests"].post('cmsswversions', cmsswVersions)[0]['result']
 
-    def getWMAgentConfig(self):
-        return self._getDataFromMemoryCache('wmagentconfig')
+    def getWMAgentConfig(self, agentName):
+        """
+        retrieve agent configuration reqmgr aux db.
+        """
+        return self._getDataFromMemoryCache('wmagentconfig/%s' % agentName)
+
+    def postWMAgentConfig(self, agentName, agentConfig):
+
+        return self["requests"].post('wmagentconfig/%s' % agentName, agentConfig)[0]['result']
+    
+    def isWMAgentDraining(self, agentName, agentDrainCondition=None):
```

Comment:
Now that I finished reading this module. I think we should separate getters from setters. And this method is clearly a getter (by its name), but it also uploades the couch record.

Just an idea, whatever is calling `isDrainMode` should first call a `updateDrainMode` or something like that and only then call `isDrainMode`. 
Second option would be renaming this method to `updateDrainMode` and moving the final logic to `isDrainMode`.
Third option is something better that you'll probably suggest :)